doc/releases/tentacle.rst

```diff
@@ -0,0 +1,395 @@
+
+  .. prompt:: bash #
+
+    ceph -W cephadm
+
+The upgrade can be paused or resumed with
+
+  .. prompt:: bash #
+
+    ceph orch upgrade pause  # to pause
+    ceph orch upgrade resume # to resume
+
+or canceled with
+
+.. prompt:: bash #
+
+    ceph orch upgrade stop
+
+Note that canceling the upgrade simply stops the process; there is no ability to downgrade back to Reef or Squid.
+
+Upgrading non-cephadm clusters
+------------------------------
+
+.. note::
+
+   1. If your cluster is running Reef (18.2.x) or later, you might choose
+      to first convert it to use cephadm so that the upgrade to Tentacle is automated (see above).
+      For more information, see https://docs.ceph.com/en/tentacle/cephadm/adoption/.
+
+   2. If your cluster is running Reef (18.2.x) or later, systemd unit file
+      names have changed to include the cluster fsid. To find the correct
+      systemd unit file name for your cluster, run following command:
+
+      ::
+
+        systemctl -l | grep <daemon type>
+
+      Example:
+
+      .. prompt:: bash $
+
+        systemctl -l | grep mon | grep active
+
+      ::
+
+        ceph-6ce0347c-314a-11ee-9b52-000af7995d6c@mon.f28-h21-000-r630.service                                           loaded active running   Ceph mon.f28-h21-000-r630 for 6ce0347c-314a-11ee-9b52-000af7995d6c
+
+#. Set the `noout` flag for the duration of the upgrade. (Optional, but recommended.)
+
+   .. prompt:: bash #
+
+      ceph osd set noout
+
+#. Upgrade monitors by installing the new packages and restarting the monitor daemons. For example, on each monitor host
+
+   .. prompt:: bash #
+
+      systemctl restart ceph-mon.target
+
+   Once all monitors are up, verify that the monitor upgrade is complete by looking for the `tentacle` string in the mon map. The command
+
+   .. prompt:: bash #
+
+      ceph mon dump | grep min_mon_release
+
+   should report:
+
+   .. prompt:: bash #
+
+      min_mon_release 20 (tentacle)
+
+   If it does not, that implies that one or more monitors hasn't been upgraded and restarted and/or the quorum does not include all monitors.
+
+#. Upgrade `ceph-mgr` daemons by installing the new packages and restarting all manager daemons. For example, on each manager host,
+
+   .. prompt:: bash #
+
+      systemctl restart ceph-mgr.target
+
+   Verify the `ceph-mgr` daemons are running by checking `ceph -s`:
+
+   .. prompt:: bash #
+
+      ceph -s
+
+   ::
+
+     ...
+       services:
+        mon: 3 daemons, quorum foo,bar,baz
+        mgr: foo(active), standbys: bar, baz
+     ...
+
+#. Upgrade all OSDs by installing the new packages and restarting the ceph-osd daemons on all OSD hosts
+
+   .. prompt:: bash #
+
+      systemctl restart ceph-osd.target
+
+#. Upgrade all CephFS MDS daemons. For each CephFS file system,
+
+   #. Disable standby_replay:
+
+         .. prompt:: bash #
+
+            ceph fs set <fs_name> allow_standby_replay false
```

Comment:
This block is indented 3 spaces further than other paragraphs and as a result is also rendered indented more than the other paragraphs in this section.